taipy/core/submission/_submission_manager.py

```diff
@@ -47,6 +53,88 @@ def _create(cls, entity_id: str, entity_type: str, entity_config: Optional[str],
+                submission._pending_jobs.discard(job.id)
+            elif job_status == Status.PENDING or job_status == Status.SUBMITTED:
+                submission._pending_jobs.add(job.id)
+                submission._blocked_jobs.discard(job.id)
+            elif job_status == Status.RUNNING:
+                submission._running_jobs.add(job.id)
+                submission._pending_jobs.discard(job.id)
+            elif job_status == Status.COMPLETED or job_status == Status.SKIPPED:
+                submission._is_completed = True  # type: ignore
+                submission._blocked_jobs.discard(job.id)
+                submission._pending_jobs.discard(job.id)
+                submission._running_jobs.discard(job.id)
+            elif job_status == Status.ABANDONED:
+                submission._is_abandoned = True  # type: ignore
+                submission._running_jobs.discard(job.id)
+                submission._blocked_jobs.discard(job.id)
+                submission._pending_jobs.discard(job.id)
+            cls._set(submission)
+
+            # The submission_status is set later to make sure notification for updating
+            # the submission_status attribute is triggered
+            if submission._is_canceled:
+                cls._set_submission_status(submission, SubmissionStatus.CANCELED, job)
+            elif submission._is_abandoned:
+                cls._set_submission_status(submission, SubmissionStatus.UNDEFINED, job)
+            elif submission._running_jobs:
+                cls._set_submission_status(submission, SubmissionStatus.RUNNING, job)
+            elif submission._pending_jobs:
+                cls._set_submission_status(submission, SubmissionStatus.PENDING, job)
+            elif submission._blocked_jobs:
+                cls._set_submission_status(submission, SubmissionStatus.BLOCKED, job)
+            elif submission._is_completed:
+                cls._set_submission_status(submission, SubmissionStatus.COMPLETED, job)
+            else:
+                cls._set_submission_status(submission, SubmissionStatus.UNDEFINED, job)
+            cls.__logger.debug(
+                f"{job.id} status is {job_status}. Submission status set to `{submission._submission_status}`"
+            )
+
+    @classmethod
+    def _set_submission_status(cls, submission: Submission, new_submission_status: SubmissionStatus, job: Job):
+        if not submission._is_in_context:
+            submission = cls._get(submission)
+        _current_submission_status = submission._submission_status
+        submission._submission_status = new_submission_status
+
+        cls._set(submission)
+
+        if _current_submission_status != submission._submission_status:
+            event = _make_event(
+                submission,
+                EventOperation.UPDATE,
+                "submission_status",
+                submission._submission_status,
+                job_triggered_submission_status_changed=job.id,
```

Comment:
This parameter name is exposed to the user through the event. It becomes an attribute of the event. We must find a better name. What about: "job_id"?

Reply:
a bit vague imho, how about `trigger_job_id`?

Reply:
I agree